client/app/scripts/superdesk/itemList/itemList.js

```diff
@@ -0,0 +1,282 @@
+            this.options = _.extend(this.options, options);
+            return this;
+        };
+
+        ItemList.prototype.addListener = function(listener) {
+            this.listeners.push(listener);
+            return this;
+        };
+
+        ItemList.prototype.removeListener = function(listener) {
+            _.remove(this.listeners, function(i) {
+                return i === listener;
+            });
+            return this;
+        };
+
+        ItemList.prototype.fetch = function() {
+            var self = this;
+
+            return itemListService.fetch(this.options)
+            .then(function(result) {
+                self.result = result;
+                self.maxPage = Math.ceil(result._meta.total / self.options.pageSize) || 0;
+                _.each(self.listeners, function(listener) {
+                    listener(result);
+                });
+                return result;
+            });
+        };
+
+        return ItemList;
+    }];
+})
+.directive('sdItemListWidget', ['ItemList', 'preferencesService', function(ItemList, preferencesService) {
+    return {
+        scope: {
+            options: '=',
+            itemListOptions: '=',
+            actions: '='
+        },
+        templateUrl: 'scripts/superdesk/itemList/views/item-list-widget.html',
+        link: function(scope, element, attrs) {
+            scope.items = null;
+            scope.maxPage = 1;
+            scope.pinnedItems = [];
+            scope.selected = null;
+
+            var pinnedList = {};
+            var itemList = new ItemList();
+            
+            itemList.addListener(function() {
+                scope.maxPage = itemList.maxPage;
+                scope.items = itemList.result;
+                processItems();
+            });
+
+            scope.$watch('itemListOptions', function() {
+                itemList.setOptions(scope.itemListOptions);
+                refresh();
+            }, true);
+
+            scope.$watch('options', function() {
+                if (scope.options.pinEnabled) {
+                    loadPinned();
+                }
+            }, true);
+
+            var _refresh = function() {
+                itemList.fetch();
+            };
+            var refresh = _.debounce(_refresh, 100);
+
+            scope.view = function(item) {
+                scope.selected = item;
+            };
+
+            scope.toggleItemType = function(itemType) {
+                if (scope.itemListOptions.types.indexOf(itemType) > -1) {
+                    scope.itemListOptions.types = _.without(scope.itemListOptions.types, itemType);
+                } else {
+                    scope.itemListOptions.types.push(itemType);
+                }
+            };
+
+            scope.isItemTypeEnabled = function(itemType) {
+                return scope.itemListOptions.types.indexOf(itemType) > -1;
+            };
+
+            scope.pin = function(item) {
```

Comment:
I'm just wondering how you get only pinned items for ingest in ingest. can't find any filtering..